src/universal/modules/teamDashboard/components/TeamSettings/TeamSettings.js

```diff
@@ -3,94 +3,204 @@ import withStyles from 'universal/styles/withStyles';
 import {css} from 'aphrodite-local-styles/no-important';
 import appTheme from 'universal/styles/theme/appTheme';
 import {overflowTouch} from 'universal/styles/helpers';
-import ui from 'universal/styles/ui';
 import {reduxForm} from 'redux-form';
-import DashSectionHeading from 'universal/components/Dashboard/DashSectionHeading';
+import InviteUser from 'universal/components/InviteUser/InviteUser';
+import UserRow from 'universal/components/UserRow/UserRow';
+import fromNow from 'universal/utils/fromNow';
+import {cashay} from 'cashay';
+import {toggleLeaveModal, toggleRemoveModal, togglePromoteModal} from 'universal/modules/teamDashboard/ducks/teamSettingsDuck';
+import RemoveTeamMemberModal from 'universal/modules/teamDashboard/components/RemoveTeamMemberModal/RemoveTeamMemberModal';
+import PromoteTeamMemberModal from 'universal/modules/teamDashboard/components/PromoteTeamMemberModal/PromoteTeamMemberModal';
+import LeaveTeamModal from 'universal/modules/teamDashboard/components/LeaveTeamModal/LeaveTeamModal';
 
 const TeamSettings = (props) => {
-  const {styles} = props;
+  const {
+    dispatch,
+    invitations,
+    leaveTeamModal,
+    modalPreferredName,
+    modalTeamMemberId,
+    myTeamMember,
+    promoteTeamMemberModal,
+    removeTeamMemberModal,
+    team,
+    teamMembers,
+    styles
+  } = props;
+  const teamLeadObj = teamMembers.find((m) => m.isLead === true);
+  const teamLead = teamLeadObj && teamLeadObj.preferredName;
+  const invitationRowActions = (invitation) => {
+    const cashayOptions = {
+      variables: {
+        inviteId: invitation.id
+      }
+    };
+    const resend = () => {
+      cashay.mutate('resendInvite', cashayOptions);
+    };
+    const cancel = () => {
+      cashay.mutate('cancelInvite', cashayOptions);
+    };
+    return (
+      <div className={css(styles.actionLinkBlock)}>
+        <div className={css(styles.actionLink)} onClick={resend}>
+          Resend Invitation
+        </div>
+        <div className={css(styles.actionLink)} onClick={cancel}>
+          Cancel Invitation
+        </div>
+      </div>
+    );
+  };
+  const teamMemberRowActions = (teamMember) => {
+    const {id, preferredName} = teamMember;
+    const openRemoveModal = () => {
+      dispatch(toggleRemoveModal(id, preferredName));
+    };
+    const openPromoteModal = () => {
+      dispatch(togglePromoteModal(id, preferredName));
+    };
+    const openLeaveModal = () => {
+      dispatch(toggleLeaveModal(id));
+    };
+    return (
+      <div className={css(styles.actionLinkBlock)}>
+        {removeTeamMemberModal &&
```

Comment:
I've erred  on the side of being more repetitive for modals so that they'll be much more flexible in the future. Still not sure I like storing 3 booleans over 1 string...